squad/core/models.py

```diff
@@ -1013,7 +1013,6 @@ def create_or_update(cls, build):
             status.metrics_summary = metrics_summary.value
             status.has_metrics = metrics_summary.has_metrics
             status.last_updated = now
-            finished, _ = build.finished
```

Comment:
I think you also need to change next line to copy finished status from Build object. Otherwise it will not change ever.

Reply:
I removed this line because `build.finished` is already called above (https://github.com/Linaro/squad/blob/master/squad/core/models.py#L996). Is this second call on line 1026 needed for some special case? I noticed build.finished computes the finished state for every call.

Reply:
Correct. I didn't look high enough :)